PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,26 @@
+### IMPORTANT: Please do not create a Pull Request without creating an issue first.
+
```

Comment:
```suggestion
```

This strikes me as an unnecessary speed bump when one identifies a bug and immediately crafts a fix.

Reply:
Although it may seem that way and I used to hold the same opinion about this, we must really ask why the process is currently so broken that someone would ever be in a situation where they had to exercise the power to immediately task-switch, diagnose, prioritize, plan, and complete a bug fix without involving the rest of the team in the issue triage, prioritization, design, refinement, work selection process.

If you're already comfortable doing all of these things on your own, create the issue for context and fill it out, it takes 5 minutes or less. It will greatly help your reviewers with understanding the what/why/how about the bug and clearly separate requirements from the work completed to meet those requirements. 

We are all human and our interpretation of requirements can vary greatly, so keeping these artifacts separate strives to reduce ambiguity.

It also allows someone else to catch that the bug has been reported and that someone is working on it so there is no duplication of effort.

This gets into a deeper conversation about task-switching and work prioritization which we will have in due time.

Reply:
It will also create issues that can be tracked as bodies of work on the [Projects board]( https://github.com/orgs/nasa-gcn/projects/6/views/1). That creates transparency and history.

Reply:
Overall, @Kirill-Vorobyev and @Courey, I agree with you. However, I think that we can skip that step for small or obvious bug fixes and typographical corrections.

Reply:
This is something I am unwilling to compromise on and is foundational to enacting a solid, transparent, and trustworthy SWE process. Every PR **MUST** have an associated issue. Every Issue **MUST** be prioritized and refined by the team before work can begin. There are many other problems with our process but we don't have the time to discuss them all here.

Reply:
I appreciate and respect that you have a strongly held conviction on this. I do as well, but in the opposite direction. I am prepared for you to convince me of your point of view, but it is not up to you alone to declare what we must do.

There is more than one valid software engineering tradition. I have certainly encountered many projects that have such a rule, and many others that don't. I think that the main predictor is whether the project is mainly corporate or academic. For example, numpy/scipy/matplotlib/astropy don't do this.

Reply:
Again, it's about traceability and accountability. If a developer creates "an obvious bug fix" and then gets hit by a bus with no documentation about what the fix was supposed to resolve in excruciating detail, then auditing the work that was done becomes much more difficult than it needs to be. What if that "obvious fix" also causes follow up issues or bugs in other places? What if it's only obvious to one person? No one can read minds, we must explicitly state all assumptions and requirements before work is started. Pull requests are also not the place to suggest how work should have been done to force a re-work. That is supposed to happen before work starts during planning and refinement.  I don't have time to teach you the entire process in this comment, this is only a stepping stone.

I am not deciding this by myself, this is the result of collaborating with @Courey and @dakota002. It is the direct result of the discussions which you yourself have sought out for us to accomplish and simultaneously chosen to exclude yourself from in your slack post.

Please see the attached issue for more details, read the acceptance criteria which describes what qualifications this work should be judged by, and please refrain from derailing this PR any further.